frontend/frontend/protocols/ssh.py

```diff
@@ -57,30 +59,100 @@ def check_channel_shell_request(self, channel: paramiko.Channel) -> bool:
     def check_channel_pty_request(self, _: paramiko.Channel, term: bytes,
                                   width: int, height: int, pixelwidth: int,
                                   pixelheight: int, modes: bytes) -> bool:
-        print("term:", term.decode("utf-8"))
-        print("width:", width)
-        print("height:", height)
-        print("pixelwidth:", pixelwidth)
-        print("pixelheight:", pixelheight)
-        # print(type(modes).__name__)
-        # print("modes:", modes.decode("utf-8"))
-        # Allow everything
+        self._session.log_pty_request(term.decode("utf-8"), width, height, pixelwidth, pixelheight)
         return True
 
 
-# todo should be singleton if the class does what it says in the docstring
-class ConnectionManager():
-    """ConnectionManager contains logic for connecting incoming
-    SSH connections to instances of Server"""
-
+class ConnectionHandler(threading.Thread):
```

Comment:
Not sure if Paramiko supports that, but you may want to check on coroutines to get that little extra edge :)